engines/rbd.c

```diff
@@ -227,12 +227,30 @@ static int _fio_rbd_connect(struct thread_data *td)
 		goto failed_shutdown;
 	}
 
+        if (td->o.odirect) {
+		r = rados_conf_set(rbd->cluster, "rbd_cache", "false");
+		if (r < 0) {
+			log_info("failed to disable RBD in-memory cache\n");
+		}
+	}
+
 	r = rbd_open(rbd->io_ctx, o->rbd_name, &rbd->image, NULL /*snap */ );
 	if (r < 0) {
 		log_err("rbd_open failed.\n");
 		goto failed_open;
 	}
 
+	if (!td->o.odirect) {
+		/*
+		 * ensure cache enables writeback/around mode unless explicitly
+		 * configured for writethrough mode
+		 */
+		r = rbd_flush(rbd->image);
```

Comment:
Should this just call fio_rbd_invalidate to be safe?

Reply:
Hmm, I am probably not understanding the concern but it might be just due to my lack of understanding the `fio` internals. At this point, since `librbd` has just opened the image, there wouldn't be anything to invalidate since we don't have the equivalent of a global kernel pagecache-like buffer shared between images.

Reply:
Ah sorry I mixed up flush with invalidate.